### PR TITLE
Feature/js api navigator

### PR DESCRIPTION
**Pull request by iNewLegend**
Dependency: 
https://github.com/elementor/elementor/pull/12025
https://github.com/elementor/elementor/pull/12099

### CODE REVIEW

File: assets/dev/js/editor/elements/views/base.js [hunk 1]
```diff
@@ -212,8 +212,7 @@ BaseElementView = BaseContainer.extend( {
 		}
 
 		this.listenTo( editModel.get( 'editSettings' ), 'change', this.onEditSettingsChanged )
```
Comment (iNewLegend) [inline assets/dev/js/editor/elements/views/base.js]: avoid

File: assets/dev/js/editor/elements/views/base.js [hunk 2]
```diff
@@ -212,8 +212,7 @@ BaseElementView = BaseContainer.extend( {
...
-			.listenTo( this.model, 'request:edit', this.onEditRequest )
-			.listenTo( this.model, 'request:toggleVisibility', this.toggleVisibility );
+			.listenTo( this.model, 'request:edit', this.onEditRequest );
```
Comment (iNewLegend) [inline assets/dev/js/editor/elements/views/base.js]: avoid

File: assets/dev/js/editor/regions/navigator/elements/commands/toggle-floding.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+		if ( undefined === state ) {
+			// TODO: move to UI HOOK.
+			element.ui.item.toggleClass( 'elementor-active', state );
+
+			element.ui.elements.slideToggle( 300, callback );
+		} else if ( state ) {
+			$e.run( 'navigator/elements/expand', newArgs );
+		} else {
+			$e.run( 'navigator/elements/collapse', newArgs );
+		}
```
Comment (iNewLegend) [inline assets/dev/js/editor/regions/navigator/elements/commands/toggle-floding.js:39]:
```suggestion
		if ( undefined === state ) {
			// TODO: move to UI HOOK.
			element.ui.item.toggleClass( 'elementor-active', state );

			element.ui.elements.slideToggle( 300, callback );
		} else if ( state ) {
			$e.run( 'navigator/elements/expand', newArgs );
		} else {
			$e.run( 'navigator/elements/collapse', newArgs );
		}
```

@matipojo, looks like this block should be **hook**?

File: assets/dev/js/editor/regions/navigator/elements/hooks/ui/hide.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import After from 'elementor-api/modules/hooks/ui/after';
```
Comment (iNewLegend) [inline assets/dev/js/editor/regions/navigator/elements/hooks/ui/hide.js:1]: Move hooks to new hooks format ( Like kit ).

File: assets/dev/js/editor/regions/navigator/commands/index.js [hunk 5]
```diff
@@ -1,3 +1,5 @@
+// Alphabetical order.
+
 export { Close } from './close';
 export { Open } from './open';
 export { Toggle } from './toggle';
```
Comment (iNewLegend) [inline assets/dev/js/editor/regions/navigator/commands/index.js:5]: test

File: assets/dev/js/editor/regions/navigator/elements/commands/index.js [hunk 6]
```diff
@@ -0,0 +1,9 @@
+// Alphabetical order.
+
+export { Expand } from './expand';
+export { Collapse } from './collapse';
+export { Hide } from './hide';
+export { Show } from './show';
+export { ToggleFolding } from './toggle-floding';
+export { ToggleFoldingAll } from './toggle-folding-all';
+export { ToggleVisibility } from './toggle-visibility';
```
Comment (iNewLegend) [inline assets/dev/js/editor/regions/navigator/elements/commands/index.js:9]: test

File: assets/dev/js/editor/regions/navigator/elements/hooks/ui/index.js [hunk 7]
```diff
@@ -0,0 +1,4 @@
+// Alphabetical order.
+
+export { NavigatorHide } from './hide';
+export { NavigatorShow } from './show';
```
Comment (iNewLegend) [inline assets/dev/js/editor/regions/navigator/elements/hooks/ui/index.js:4]: test

File: assets/dev/js/editor/regions/navigator/navigator.js [hunk 8]
```diff
@@ -1,14 +1,13 @@
-import Component from './component';
+import NavigatorComponent from './component';
```
Comment (iNewLegend) [inline assets/dev/js/editor/regions/navigator/navigator.js:1]: Test whole behavior?

File: assets/dev/js/editor/regions/navigator/commands/index.js (hunk 5 above)
Comment (iNewLegend) [reply assets/dev/js/editor/regions/navigator/commands/index.js]: Handled: https://github.com/elementor/elementor/pull/10132/commits/95c62be8a3317a79f7c6d453cb0a18e581f6b65b

File: assets/dev/js/editor/elements/views/base.js [hunk 9]
```diff
@@ -319,12 +318,6 @@ BaseElementView = BaseContainer.extend( {
 		return 'content' !== control.tab || control.selectors || control.prefix_class;
 	},
 
-	toggleVisibility() {
```
Comment (iNewLegend) [inline assets/dev/js/editor/elements/views/base.js:322]: BC?

File: assets/dev/js/editor/regions/navigator/layout.js [hunk 10]
```diff
@@ -29,18 +29,6 @@ export default class extends Marionette.LayoutView {
...
-		const state = 'expand' === this.ui.toggleAll.data( 'elementor-action' ),
-			classes = [ 'eicon-collapse', 'eicon-expand' ];
-
-		this.ui.toggleAll
-			.data( 'elementor-action', state ? 'collapse' : 'expand' )
-			.removeClass( classes[ +state ] )
-			.addClass( classes[ +! state ] );
-
-		this.elements.currentView.recursiveChildInvoke( 'toggleList', state );
-	}
```
Comment (iNewLegend) [inline assets/dev/js/editor/regions/navigator/layout.js:42]: BC?

**Comment by matipojo** [comment]
It will be merged in the future.

**Pull request closed without merging**